src/engraving/libmscore/barline.cpp

```diff
@@ -1137,9 +1153,10 @@ void BarLine::endEditDrag(EditData& ed)
     int newSpanFrom = 0;
     int newSpanTo = 0;
 
-    bool localDrag = ed.control() || segment()->isBarLineType();
+    bool localDrag = ed.control() || segment()->isBarLineType() || spanStaff() != score()->staff(staffIdx())->barLineSpan();
```

Comment:
fixes a bug (also in MU3) where trying to adjust a barline previously adjusted using "CTRL" key modifier doesn't work